### PR TITLE
Queue contacts updates

### LINKED ISSUE #842: Editing nearly impossible - spinning wheel and "The contact you were trying to edit has changed" messages"
### Describe the bug
  Editing of fields in the mask lead to "changed" messages and the edited text will be discarded

**To Reproduce**
When adding a new contact or editing an existing one, I go into any of the editing fields and try to enter some text there.
A soon as I'm "slow" in editing, the spinner wheel in the top right corner of the mask appears, I assume to show me that it tries to sync the entered data with the data.
When I continue typing and don't wait until the spinning wheel disappears - the message "_The contact you were trying to edit has changed. Please manually refresh the contact. Any further edits will be discarded_" appears.
When I then click on that button to refresh the data one of the two following things happen:
  1. my entered / edited data get's cut somewhere and characters are missing and changes are rejected or
  2. everything seems to be fine (texts are completely shown) until I change to another contact and then back to the initial contact - then the entered data fields miss some of the changes

**Expected behavior**
Editing of text in the fields should be kept back until the input field will be changed - and then the spinning wheel may appear and transfer the changed text to the server.

**Actual behavior**
The spinning wheel occurs when the typing is not "fast" enough - it seems that it takes some time to transfer the data to the server; when typing / more changes occur in that time the message seems to occur...

### Server configuration
The mentioned Issue Template app is not available for Nextcloud 15...

**Operating system**:
  Server: Linux (Debian, Shared Hoster)
  Client: Windows 10

**Web server:** Apache

**Database:** MariaDB 5.5.31

**PHP version:** 7.1.2

**Nextcloud version:** 15.0

**Contacts version:** 3.0.1

**Updated from an older Nextcloud or fresh install:** Update from latest V14

**Signing status:**
```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```

**List of activated apps:**
`  - activity: 2.8.2
  - admin_audit: 1.5.0
  - announcementcenter: 3.4.0
  - apporder: 0.6.0
  - audioplayer: 2.5.1
  - bookmarks: 0.15.1
  - calendar: 1.6.4
  - cloud_federation_api: 0.1.0
  - comments: 1.5.0
  - contacts: 3.0.1
  - dav: 1.8.0
  - external: 3.2.0
  - federatedfilesharing: 1.5.0
  - federation: 1.5.0
  - files: 1.10.0
  - files_external: 1.6.0
  - files_pdfviewer: 1.4.0
  - files_sharing: 1.7.0
  - files_texteditor: 2.7.0
  - files_trashbin: 1.5.0
  - files_versions: 1.8.0
  - files_videoplayer: 1.4.0
  - firstrunwizard: 2.4.0
  - flowupload: 0.0.9
  - gallery: 18.2.0
  - gpxpod: 3.0.1
  - groupfolders: 2.0.2
  - impersonate: 1.2.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.3.0
  - metadata: 0.8.0
  - nextcloud_announcements: 1.4.0
  - notifications: 2.3.0
  - oauth2: 1.3.0
  - polls: 0.9.5
  - previewgenerator: 2.0.0
  - provisioning_api: 1.5.0
  - serverinfo: 1.5.0
  - sharebymail: 1.5.0
  - systemtags: 1.5.0
  - tasks: 0.9.8
  - theming: 1.6.0
  - twofactor_backupcodes: 1.4.1
  - updatenotification: 1.5.0
  - user_usage_report: 1.1.1
  - workflowengine: 1.5.0`

**Nextcloud configuration:**
`{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***",
            "***REMOVED SENSITIVE VALUE***"
        ],
        "activity_expire_days": "60",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "filelocking.debug": false,
        "filelocking.enabled": true,
        "filelocking.ttl": 1800,
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "version": "15.0.0.10",
        "installed": true,
        "dbtype": "mysql",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "default_language": "de",
        "defaultapp": "files",
        "integrity.check.disabled": true,
        "knowledgebaseenabled": true,
        "enable_avatars": true,
        "allow_user_to_change_display_name": true,
        "theme": "",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "loglevel": 3,
        "logdateformat": "Y-m-d H:i:s",
        "logtimezone": "Europe\/Berlin",
        "log_rotate_size": 20971520,
        "skeletondirectory": "\/home\/www\/nextcloud_data\/skeleton",
        "updatechecker": true,
        "trashbin_retention_obligation": "auto",
        "versions_retention_obligation": "disabled",
        "appstoreenabled": true,
        "appstore.experimental.enabled": true,
        "singleuser": false,
        "maintenance": false,
        "htaccess.RewriteBase": "\/",
        "mail_smtpmode": "smtp",
        "updater.release.channel": "stable",
        "mail_smtpsecure": "ssl",
        "mail_smtpauthtype": "LOGIN",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauth": 1,
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "updater.secret": "***REMOVED SENSITIVE VALUE***",
    }`

### Client configuration
**Browser:** Firefox 64.0, 64-Bit

**Operating system:**

Windows 10 - Pro

**CardDAV-clients:**

#### *Web server error log*
```
Insert your webserver log here
```

#### *data/nextcloud.log*
```
Insert your Nextcloud log here
```

#### Browser log

Console Log:
```
JQMIGRATE: Migrate is installed, version 1.4.0 core.js:7:542
New search handler registered search.js:67:4
Connected to dav! 
Object { rootUrl: Getter & Setter, advertisedFeatures: Getter & Setter, currentUserPrincipal: Getter & Setter, principalCollections: Getter & Setter, calendarHomes: Getter & Setter, addressBookHomes: Getter & Setter, parser: Getter & Setter, _isConnected: Getter & Setter, _request: Getter & Setter, … }
contacts.js:84:29413
Source-Map-Fehler: TypeError: NetworkError when attempting to fetch resource.
Ressourcen-Adresse: https://***REMOVED SENSITIVE VALUE***/apps/contacts/js/contacts.js?v=5865a6bf-48
Source-Map-Adresse: contacts.js.map[Weitere Informationen]
```
netzwerklog

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/68476419-editing-nearly-impossible-spinning-wheel-and-the-contact-you-were-trying-to-edit-has-changed-messages?utm_campaign=plugin&utm_content=tracker%2F46751899&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F46751899&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by skjnldsv**
@georgehrke 

Fix #842

### CODE REVIEW

File: src/components/ContactDetails.vue [hunk 1]
```diff
@@ -308,7 +311,7 @@ export default {
 			},
 			set: function(data) {
 				this.contact.groups = data
-				this.updateContact()
+				this.debounceUpdateContact()
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply src/components/ContactDetails.vue]: I guess yes! :)
It doesn't harms to debounce anyway ;)

**Comment by georgehrke** [comment]
@ningappa How did you test this? This fix wasn't released in a  stable version yet.

**Comment by ningappa** [comment]
@georgehrke comment is specific to this [commit](https://github.com/nextcloud/contacts/pull/879/commits/eae3c151b296217b4be1338db0e8dab26ece02ba).
I've installed test nextcloud locally. I do watch daily commits of contact app and I take those changes to local. I hope I'm not doing anything wrong.

**Comment by ningappa** [comment]
@skjnldsv I changed bit of webpack common configuration file. Everytime i build, it create new file with hash in the filename and i used `find-replace-webpack-plugin` to generate dynamic main.php . So there is no need of clear cache since every build loads new application js file to the application.

**Pull request merged**